File: server/common/models/player.py
from django.db import models
from django.core import validators
from common.models import Base

class Player(Base):
    # generic attributes
    first_name = models.CharField(max_length=64)
    last_name = models.CharField(max_length=64)

    # offensive attributes
    passing = models.PositiveSmallIntegerField(validators=[validators.MaxValueValidator(3)])
    hands = models.PositiveSmallIntegerField(validators=[validators.MaxValueValidator(20)])
    rushing = models.PositiveSmallIntegerField(validators=[validators.MaxValueValidator(5)])
    receive_short = models.PositiveSmallIntegerField(validators=[validators.MaxValueValidator(5)])
    receive_mid = models.PositiveSmallIntegerField(validators=[validators.MaxValueValidator(5)])
    receive_deep = models.PositiveSmallIntegerField(validators=[validators.MaxValueValidator(5)])
    blocking = models.PositiveSmallIntegerField(validators=[validators.MaxValueValidator(5)])
    kicking = models.PositiveSmallIntegerField(validators=[validators.MaxValueValidator(5)])

    # offensive_special = models.ManyToManyField(SpecialAbility, limit_choices_to={'type': 'offense'})

    pass_man = models.PositiveSmallIntegerField(validators=[validators.MaxValueValidator(3)])
    pass_zone = models.PositiveSmallIntegerField(validators=[validators.MaxValueValidator(5)])
    run_defense = models.PositiveSmallIntegerField(validators=[validators.MaxValueValidator(5)])
    blitzing = models.PositiveSmallIntegerField(validators=[validators.MaxValueValidator(5)])

    # offensive_special = models.ManyToManyField(SpecialAbility, limit_choices_to={'type': 'defense'})

    def __unicode__(self):
        return "{f} {l}".format(f=self.first_name, l=self.last_name)

File: server/common/static/common/models/player.js
define([
        'common/js/util',
        'common/js/model'
    ], function(
        util,
        Base
    ) {
        function Player() {
            var self = this;

            self.constructor = Player;

            Base.call(self);
        }

        Player.info = {
            name: 'Player',
            fields: {
                //generic
                'first_name': 'string',
                'last_name': 'string',

                //offensive
                'passing': 'number',
                'hands': 'number',
                'rushing': 'number',
                'receive_short': 'number',
                'receive_mid': 'number',
                'receive_deep': 'number',
                'blocking': 'number',
                'kicking': 'number',

                'offensive_special': 'array',

                //defensive
                'pass_man': 'number',
                'pass_zone': 'number',
                'run_defense': 'number',
                'blitzing': 'number',

                'defensive_special': 'array',
            }
        };

        return Player;
    }
);
